packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -112,7 +112,7 @@ const createHistoryService = ({ strapi }: { strapi: LoadedStrapi }) => {
           where: {
             $and: [
               { contentType: params.contentType },
-              { relatedDocumentId: params.documentId },
+              ...(params.documentId ? [{ relatedDocumentId: params.documentId }] : []),
               ...(params.locale ? [{ locale: params.locale }] : []),
             ],
           },
```

Comment:
shouldn't you provide the publication state of the document? wether you want the draft or the published version?

Reply:
I think we want to return all history-versions for a document including both draft and published states. 🤔